src/libraries/Common/src/System/Security/Cryptography/Oids.Shared.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Security.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    internal static partial class Oids
+    {
+        internal static readonly Oid RsaOid = new Oid(Rsa, "RSA");
+        internal static readonly Oid EcPublicKeyOid = new Oid(EcPublicKey, "ECC");
+        internal static readonly Oid secp256r1Oid = new Oid(secp256r1, nameof(ECCurve.NamedCurves.nistP256));
+        internal static readonly Oid secp384r1Oid = new Oid(secp384r1, nameof(ECCurve.NamedCurves.nistP384));
+        internal static readonly Oid secp521r1Oid = new Oid(secp521r1, nameof(ECCurve.NamedCurves.nistP521));
```

Comment:
As this gets bigger it'll be better to turn them into lazy-init properties, so may as well do that now.

```C#
private static readonly Oid _rsaOid;
private static readonly Oid _secp521r1Oid;

internal static Oid RsaOid => _rsaOid ??= MakeOid(Rsa);
internal static Oid secp521r1Oid => _secp521r1Oid ??= new Oid(secp521r1, nameof(ECCurve.NamedCurves.nistP521));

private static Oid MakeSharedOid(string oidValue)
{
    Oid oid = new Oid(oidValue, null);
    // do not remove this variable read, the property get has a side effect.
    string? friendlyName = oid.FriendlyName;
    Debug.Assert(friendlyName != null, $"Shared OID {oidValue} did not populate FriendlyName");
    return oid;
}
```